src/common.ts

```diff
@@ -34,3 +35,16 @@ export function getLastTimestamp(): number {
 	const file = JSON.parse(fs.readFileSync(fileToCheck).toString());
 	return file.timestamp;
 }
+
+export function tiebreakerSortFactory(timeframe: 'weekly' | 'total', walletsStats: WalletsStats) {
+	return (address_a: string, address_b: string): number => {
+		const walletStat_a = walletsStats[address_a];
+		const walletStat_b = walletsStats[address_b];
+		const volumeDiff = walletStat_a[timeframe].byteCount - walletStat_b[timeframe].byteCount;
+		const tipsDiff = walletStat_a[timeframe].tips - walletStat_b[timeframe].tips;
+		const blockSinceParticipatingDiff =
+			// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+			walletStat_a[timeframe].blockSinceParticipating! - walletStat_b[timeframe].blockSinceParticipating!;
```

Comment:
For daily/weekly participants it is. But now we are working with non-participants. And I realized the sort order was ascendant, now it's descendant as we want the bigger byte count, tips, and blockSinceParticipating first
0643b4a7b7f34d7d64ae9f9398aab157921e1e8f

Reply:
**Edit**
Actually, the blockSinceParticipating should be sorted in ascendant order. Because we want the smaller block first
47aaa2ac061cedc026aaa044a95e47088b0fc69e

Reply:
So in this new algorithm,
- if the values are equal (either both numbers or both undefined) the diff is zero,
- if some of the -distinct- values are undefined, then we use `Number.POSITIVE_INFINITY` to compare with, and the method will return some of `Number.POSITIVE_INFINITY` or `Number.NEGATIVE_INFINITY`.
- if both are numbers, then the difference is returned

Reply:
Since this is now a pure function and was sorting the wrong direction, this might be a good opportunity to unit test the function to find any hidden details and for future regressions